pyharmonysearch/harmony_search.py

```diff
@@ -126,32 +133,50 @@ def run(self):
                 else:
                     self._random_selection(harmony, i)
             fitness = self._obj_fun.get_fitness(harmony)
-
             self._update_harmony_memory(harmony, fitness)
-
             num_imp += 1
 
+            # save harmonies every n_inds th improvisations (i.e., one 'generation')
+            if num_imp % self._obj_fun.get_hms() == 0:
```

Comment:
Just to be clear, each time this conditional trips, this is the most often the entire harmony memory *could* be replaced. But it might not have changed at all. So your goal here is just to get harmony memory each time it could be completely different, right? That makes sense to me, but I just want to be clear.

I hope that made sense.

I think it'd be nice to explain this in a comment above (replace line 139 with a more detailed explanation of what this conditional is doing).

Reply:
I was actually thinking to save the harmony memory every generation, in which every individual solution is evaluated like in PSO. But here, it doesn't seem to be the case. What do you think is the most appropriate option for the user? Update every improvisation (I am interpreting this as a single fitness evaluation)? Or nth improvisations, where n is specified by the user?